www/install/php/Update-22.10.0-beta.1.php

```diff
@@ -56,3 +62,310 @@
+    $stmt->execute();
+
+    $stmt = $pearDB->query(
+        "SELECT cb_type_id, type_shortname FROM cb_type WHERE type_shortname in ('bbdo_server', 'bbdo_client')"
+    );
+
+    return $stmt->fetchAll(\PDO::FETCH_KEY_PAIR);
+}
+
+/**
+ * @param CentreonDB $pearDB
+ * @param array<string,string|int|null> $fields
+ * @return array<string,string|int|null>
+ */
+function insertFields(CentreonDB $pearDB, array $fields): array
+{
+    $fieldStmt = $pearDB->prepare(
+        "INSERT INTO cb_field (fieldname, displayname, fieldtype, description) VALUES
+        (:fieldname, :displayname, :fieldtype, :description)"
+    );
+
+    foreach ($fields as &$field) {
+        $fieldStmt->bindValue(':fieldname', $field['fieldname'], \PDO::PARAM_STR);
+        $fieldStmt->bindValue(':displayname', $field['displayname'], \PDO::PARAM_STR);
+        $fieldStmt->bindValue(':fieldtype', $field['fieldtype'], \PDO::PARAM_STR);
+        $fieldStmt->bindValue(':description', $field['description'], \PDO::PARAM_STR);
+        $fieldStmt->execute();
+
+        $field['id'] = $pearDB->lastInsertId();
+
+        if (in_array($field['fieldtype'], ['radio', 'multiselect'])) {
+            insertFieldOptions($pearDB, $field);
+        }
+    }
+
+    return $fields;
+}
+
+/**
+ * @param CentreonDB $pearDB
+ * @param array<string,string|int|null> $field
+ */
+function insertFieldOptions(CentreonDB $pearDB, array $field): void
+{
+    $fieldOptionsStmt = $pearDB->prepare(
+        "INSERT INTO cb_list (cb_list_id, cb_field_id, default_value) VALUES (:listId, :fieldId, :defaultValue)"
+    );
+    $fieldOptionsStmt->bindValue(':listId', $field['optionListId'], \PDO::PARAM_INT);
+    $fieldOptionsStmt->bindValue(':fieldId', $field['id'], \PDO::PARAM_INT);
+    $fieldOptionsStmt->bindValue(':defaultValue', $field['defaultValue'], \PDO::PARAM_STR);
+    $fieldOptionsStmt->execute();
+}
+
+/**
+ * @return array{bbdo_server:<string,string|int|null>,bbdo_client:<string,string|int|null>}
+ */
+function getFieldsDetails(): array
+{
+    $bbdoServer = [
+        [
+            "fieldname" => 'host',
+            "displayname" => 'Listening address (optional)',
+            "fieldtype" => 'text',
+            "description" =>  'Fill in this field only if you want to specify the address on which Broker '
+                . 'should listen',
+            "isRequired" => 0
+        ],
+        [
+            "fieldname" => 'port',
+            "displayname" => 'Listening port',
+            "fieldtype" => 'text',
+            "description" => 'TCP port on which Broker should listen',
+            "isRequired" => 1
+        ],
+        [
+            "fieldname" => 'transport_protocol',
+            "displayname" => 'Transport protocol',
+            "fieldtype" => 'radio',
+            "description" => 'The transport protocol can be either TCP (binary flow over TCP) or gRPC (HTTP2)',
+            "isRequired" => 1,
+            "defaultValue" => 'gRPC',
+            "optionListId" => 13,
```

Comment:
you should not use hardcoded ids, cause some modules can add broker configuration and increment ids
(example : https://github.com/centreon/centreon-bam/blob/develop/www/modules/centreon-bam-server/tools/installBrokerConf.php#L90)